### ISSUE TITLE
Issue with Sofort payments

### ISSUE THREAD

**Issue by jorgb90** [mollie/magento2#180]
We had several Sofort payments on our webshop which resulted in chargebacks and because of that are not paid out by Mollie. Now it just came to our understanding that Sofort is not a guaranteed payment method like iDeal (while its advertised the same by Mollie and others), but its only guaranteed when Mollie has received the payment on their bankaccount. Now the big issue comes and thats that we live in a world where shipments are expected within 24 hours and Mollie's dashboard or this plugin tells us the payment is Paid while in fact it isn't, by telling us the order is paid the items are of course shipped. We do not have any other way to determine the "guaranteed" status. Shouldn't Sofort payments be set to paid until the funds are received by Mollie and NOT before that? Cause now we (the shops) are the ones who lose money..

Long story short.. Give Sofort payments not a Paid status until the funds are received by Mollie and payout can be guaranteed!

**Comment by Frank-Magmodules** [comment]
Hi @jorgb90 , thanks for opening this issue, i will investigate this with the Mollie team and let you know once i know more on this.

**Comment by Frank-Magmodules** [comment]
Hi @jorgb90, i've talked with Mollie on this and it seems that the flow that you describe works like it should, but they use the payment state "paid" before the money is guaranteed so they recommend to only ship the order after Mollie did the payout on this order. 
Also see;
https://help.mollie.com/hc/nl/articles/115000310169-Hoe-werkt-SOFORT-Banking- 

I can imagine this isn’t an ideal flow but they are going to introduce Klarna Pay Now in the future which is going to be launched as a “Sofort 2.0” solution as they payment is guaranteed.

**Comment by jorgb90** [comment]
@Frank-Magmodules Yes thats exactly what I said. As long as "Sofort 2.0" isn't launched webshops are better not using Sofort through Mollie, since there is no control over the status of the payment. Its like saying a banktransfer is paid when the initial order is placed, which is of course not paid until the funds are received :/

**Comment by JorikMollie** [comment]
Hi @jorgb90,

As this issue is no longer related to the Magento plugin @Frank-Magmodules asked me to pitch in.

In comparison with guaranteed payment methods Sofort has the downside that a payment can get the status charged back.

As the percentage affected Sofort payments is negligible, especially compared to the percentage of failed bank transfers, we decided to give a Sofort payment the status Paid as soon as we get the first confirmation.

This is a business decision with a downside: the package is probably sent by the time the chargeback is known. If this downside of Sofort is too big for you I can understand that you disable Sofort for now and wait for the release of Klarna Now.

If you want more information about our reasons for this decision I ask you to contact us at info@mollie.com. GitHub is not the place for that exchange of views.

**Comment by jorgb90** [comment]
@JorikMollie I will not contact info@mollie.com. Just wanted to start a discussion here to see if we can change the way Sofort payments are handled. This also can be done by changing this module, so it is related. And for your information we had 4 Sofort payments of each around 4000 euros which all 4 resulted in a chargeback, thats 100% (negligible?) and 16000 euros. A very bad business decision and a perfect way for fraud!

**Comment by JorikMollie** [comment]
@jorgb90 I've created a feature request to add a configuration option to Magento to set an order paid with Sofort to Paid only after the payment has been settled.

Based on the feedback we get from within Mollie as outside Mollie we'll decide wether and when we'll implement the feature request.
